### ISSUE TITLE
How to send commands to a jupyter notebook terminal through http requests such as POST.

### ISSUE THREAD

**Issue by gpamfilis** [jupyter/notebook#3611]
I have opened a jupyter terminal at http://localhost:8888/terminals/1 is there a way to send a POST request to this url in order to run a command. for example {"command":"ls"} to see what files are there or {"command":"python runsomething.py"}. I tried doing a GET request from POSTMAN even after i included the token and it complained about javascript. Is there a way to this though because i can not seem to find it in the documentation.

EDIT

i know how to run general terminal commands (not jupyter notebook terminal)from a notebook cell using !python runsomething.py etc. What i am interested is purely an http communication with a terminal opened from jupyter.

**Comment by takluyver** [comment]
No, the terminals are run over websockets, with a protocol based on keystrokes rather than commands.

You might be interested in something like [kernel_gateway](https://github.com/jupyter/kernel_gateway).
